### PR TITLE
fix jax.random.logits shape argument

### LINKED ISSUE #13124: jax.random.categorical doesn't respect shape parameter with broadcasted logits
### Description

The docs for `jax.random.categorical` say that the output of `jax.random.categorical` is `shape` when `shape` is provided. However, when provided `logits` which have to be broadcasted with `shape`, it just ignores the shape parameter after checking that it's broadcast compatible. I believe this can be fixed by broadcasting logits appropriately.

```
import jax
out = jax.random.categorical(jax.random.PRNGKey(0), logits=jax.numpy.ones( (4, 1, 10)), axis=-1, shape=(4, 8))

# out.shape is (4, 1), should be (4, 8)

assert out.shape == (4, 8), f"out.shape is {out.shape}, should be (4, 8)"
```

### What jax/jaxlib version are you using?

0.3.23

### Which accelerator(s) are you using?

CPU

### Additional system info

_No response_

### NVIDIA GPU info

_No response_

### PR DESCRIPTION
**Pull request by mattjj**
fixes #13124

### CODE REVIEW

File: jax/_src/random.py [hunk 1]
```diff
@@ -1327,10 +1327,13 @@ def categorical(key: KeyArray,
     shape = tuple(shape)
     _check_shape("categorical", shape, batch_shape)
 
-  sample_shape = shape[:len(shape)-len(batch_shape)]
+  shape_prefix = shape[:len(shape)-len(batch_shape)]
+  logits_shape = list(shape[len(shape) - len(batch_shape):])
+  logits_shape.insert(axis % len(logits_arr.shape), logits_arr.shape[axis])
+  logits_arr= jnp.broadcast_to(logits_arr, logits_shape)
```
Comment (sharadmv) [inline jax/_src/random.py]: insert space before =

**Review by sharadmv — APPROVED** [review]

File: jax/_src/random.py (hunk 1 above)
Comment (mattjj) [reply jax/_src/random.py]: Done! Also I deleted that line just now.
